tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9776,51 +9783,60 @@ TEST_F(ExecutionTest, DynamicResourcesDynamicIndexingTest) {
     24.0, 25.0, 
     30.0, 31.0, 
     32.0, 33.0};
-  float *expectedResultsArray[2];
-  expectedResultsArray[0] = expectedResultsUniform;
-  expectedResultsArray[1] = expectedResultsNonUniform;
     
-  for (unsigned int non_uniform_bit = 0; non_uniform_bit < 2; non_uniform_bit++) {
-    for (unsigned i = 0; i < _countof(TestShaderModels); i++) {
-      D3D_SHADER_MODEL sm = TestShaderModels[i];
-      LogCommentFmt(L"\r\nVerifying Dynamic Resources Uniform Indexing in shader "
-                    L"model 6.%1u",
-                    ((UINT)sm & 0x0f));
+  // TestShaderModels will be an array, where the first x models are "non-fallback", and the rest of the models
+  // are "fallback". If TestShaderModels has length y, and a test loops through all shader models, a convention
+  // to test based on whether fallback is enabled or not is to limit the loop like this:
+  // unsigned num_models_to_test = ExecutionTest::IsFallbackPathEnabled() ? y : x;
+  unsigned num_models_to_test = ExecutionTest::IsFallbackPathEnabled() ? 2 : 1;
+  for (unsigned i = 0; i < num_models_to_test; i++) {
+    D3D_SHADER_MODEL sm = TestShaderModels[i];
+    LogCommentFmt(L"\r\nVerifying Dynamic Resources Uniform Indexing in shader "
+                  L"model 6.%1u",
+                  ((UINT)sm & 0x0f));
 
-      CComPtr<ID3D12Device> pDevice;
-      if (!CreateDevice(&pDevice, sm, false /* skipUnsupported */)) {
-        continue;
-      }
-      D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
-      VERIFY_SUCCEEDED(
-          pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
-                                       &devOptions, sizeof(devOptions)));
-      if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
-        WEX::Logging::Log::Comment(
-            L"Device does not support Resource Binding Tier 3");
-        WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
-        return;
-      }
+    CComPtr<ID3D12Device> pDevice;
+    if (!CreateDevice(&pDevice, sm, false /* skipUnsupported */)) {
+      continue;
+    }
+    D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
+    VERIFY_SUCCEEDED(
+        pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
+                                      &devOptions, sizeof(devOptions)));
+    if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
+      WEX::Logging::Log::Comment(
+          L"Device does not support Resource Binding Tier 3");
+      WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+      return;
+    }
 
-      // Add compile options
+    for (unsigned int non_uniform_bit = 0; non_uniform_bit < 2; non_uniform_bit++) {
+      float *expectedResults = non_uniform_bit ? expectedResultsNonUniform : expectedResultsUniform;
+
+      LogCommentFmt(L"\r\nnon uniform bit is %1u",
```

Comment:
Why `\r\n` in front of the message?  Plus this message isn't very consistent with conventions here.  What about:
```
LogCommentFmt(L"Testing %s Resource Indexing.", non_uniform_bit ? L"NonUniform" : L"Uniform");
```